commons-packet/commons-packet-manager/src/main/java/io/mosip/commons/packet/impl/PacketReaderImpl.java

```diff
@@ -251,7 +255,28 @@ public BiometricRecord getBiometric(String id, String biometricFieldName, List<B
             biometricRecord = new BiometricRecord();
             List<BIR> segments = new ArrayList<>();
             List<io.mosip.kernel.core.cbeffutil.entity.BIR> birList = CbeffValidator.convertBIRTypeToBIR(birType.getBIR());
-            birList.forEach(bir -> segments.add(PacketManagerHelper.convertToBiometricRecordBIR(bir)));
+            if (modalities==null || modalities.isEmpty()) {
+				birList.forEach(bir -> segments.add(PacketManagerHelper.convertToBiometricRecordBIR(bir)));
+			} else {
+				Map<String, BIR> birMap = new HashMap<>();
+
+				for (io.mosip.kernel.core.cbeffutil.entity.BIR bir : birList) {
+					String mapKey = "";
+					for (String subType : bir.getBdbInfo().getSubtype()) {
+						mapKey += subType;
```

Comment:
1. in case of face subtype wont be there. kindly make sure u do proper null checks to avoid null pointer,
2. create a method filterByModalities() and move entire filteration logic there.